posts/2020-11-09-ocp-log-forwarding.adoc

```diff
@@ -320,15 +322,36 @@ metadata:
 spec:
   managementState: "Unmanaged"
 ```
-*Note:* After you set this value to *"Unmanaged"*, any further changes to the _ClusterLogging_ or _LogForwarding_ instances are not automatically detected. To automatically detect changes, you must change the *managementState* field back to *"Managed"*.
 
-Next, set the *MERGE_JSON_LOG* environment variable to *true*.
+Alternatively, you can modify this file on the web console. Ensure you are in the `openshift-logging` project namespace and go to `Administration > Custom Resource Definitions > ClusterLogging > Instances > Instance > YAML`
+
+Next, we'll need to change the  *merge_json_log* attribute of the `kubernetes.var.log.containers.**` filter to *true* in the Fluentd's config map.
+
+```
+[root@ocp ~]# oc edit configmap fluentd
 
+apiVersion: v1
+data:
+...
+  <filter kubernetes.var.log.containers.**>
+    @type parse_json_field
+    merge_json_log 'true'
+    preserve_json_log 'true'
+    json_fields 'log,MESSAGE'
+  </filter>
+...
+```
+
+Finally we need to force restart the Fluentd pods to pick up this new change:
 [source]
 ----
-[root@ocp ~]# oc set env ds/fluentd MERGE_JSON_LOG=true
+oc delete pod --selector logging-infra=fluentd
 ----
 
+Alternatively, you can modify this file on the web console. Ensure you are in the `openshift-logging` project namespace and go to `Workloads > Config Maps > fluentd > YAML`
+
+*Note:* After you set this value to *"Unmanaged"*, any further changes to the _ClusterLogging_ or _ClusterLogForwarder_ instances are not automatically detected. To automatically detect changes, you must change the *managementState* field back to *"Managed"*. By switching back to *"Managed"* the Cluster Logging instance will revert the Fluentd's config map back to the original settings.
```

Comment:
Should this note be at line 327, after the bit about changing the management state? Or is it here because resetting the config map undoes both parts of this step? 

Also, it's unclear whether parsing JSON fields and automatically detecting changes are mutually exclusive. Is the implication that a user would have to choose between automatically detecting changes and forwarding logs to Splunk? Not sure we have to unpack that here but it does seem a little ambiguous

Reply:
Ah resetting the cluster logging instance's 'managementState` attribute back to managed gives control of the fluentd configmap back to the clusterlogging's operator? I'm not sure on the details. But once it is back to managed it takes control again. And the fluentd config map is part of it's control and it will override any changes made to it and reset it to the default. I hope the new explanation makes it a little bit more clear.